File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/puzzles/tapaislands/domain/TapaIslandsObject.java
package com.zwstudio.logicpuzzlesandroid.puzzles.tapaislands.domain;

public abstract class TapaIslandsObject {
    public abstract String objTypeAsString();
    public static TapaIslandsObject objTypeFromString(String str) {
        switch (str) {
            case "marker":
                return new TapaIslandsMarkerObject();
            case "wall":
                return new TapaIslandsWallObject();
            default:
                return new TapaIslandsEmptyObject();
        }
    }
}

File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/puzzles/tapaislands/TapaIslandsObject.kt
package com.zwstudio.logicpuzzlesandroid.puzzles.tapaislands

import com.zwstudio.logicpuzzlesandroid.common.domain.Position

enum class TapaIslandsObject {
    Empty, Hint, Marker, Wall
}

class TapaIslandsGameMove(val p: Position, var obj: TapaIslandsObject = TapaIslandsObject.Empty)
